src/documentation/io/authoring-overview.md

```diff
@@ -10,35 +10,76 @@ permalink: /documentation/io/authoring-overview/
+* **Reading from a file glob** - For example reading all files in "~/data/**"
+  * Get File Paths ParDo: As input, take in a file glob. Emit a PCollection of strings, each of which is a file path.
+  * Reading ParDo: Given the PCollection of file paths, read each one, emitting a PCollection of records.
+* **Reading from a NoSQL Database** (eg Apache HBase) - these databases often allow reading from ranges in parallel.
+  * Determine Key Ranges ParDo: As input, receive connection information for the database and the key range to read from. Output a PCollection of key ranges that can be read in parallel efficiently.
+  * Read Key Range ParDo: Given the PCollection of key ranges, read the key range, emitting a PCollection of records.
+
+For data stores or files where reading cannot occur in parallel, reading is a simple task that can be accomplished with a single ParDo. For example:
+* **Reading from a database query** - traditional SQL database queries often can only be read in sequence. The ParDo in this case would establish a connection to the database and read batches of records, emitting a PCollection of those records.
+* **Reading from a gzip file** - a gzip file has to be read in order, so it cannot be parallelized. The ParDo in this case would open the file and read in sequence, emitting a PCollection of records from the file.
+
 
 ### When to implement using the Source API
-TODO
+The above discussion is in terms of ParDos - this is because Sources have proven to be tricky to implement. At this point in time, **it's not recommended that you implement a Source if you can get away with a ParDo**. If you're trying to decide between the two, feel free to email the Beam dev mailing list and we can discuss the specific pros and cons of your case.
+
+In some cases implementing a Source may be necessary or result in better performance.
+* ParDos will not work for reading from unbounded sources - they do not support checkpointing and don't support mechanism like de-duping that have proven useful for streaming data sources.
+* ParDos cannot provide hints to runners about their progress or the size of data they are reading -  without size estimation of the data or progress on your read, the runner doesn't have any way to guess how large your read will be, and thus if it attempts to dynamically allocate workers, it does not have any clues as to how many workers you may need for you pipeline.
+* ParDos do not support Dynamic Work Rebalancing - these are features used by some readers to improve the processing speed of jobs (but may not be possible with your data source.)
+* ParDos do not receive 'desired_bundle_size' as a hint from runners when performing initial splitting.
+
 
 ## Write transforms
-TODO
+Write transforms are responsible for the work of taking the contents of a PCollection and transferring that data outside of the Beam Pipeline.
```

Comment:
i'd axe "the work of"
nit: lowercase pipeline (i think we've been lowercasing pipeline in general)